src/otx/algorithms/common/adapters/mmcv/configurer.py

```diff
@@ -345,6 +380,69 @@ def configure_compat_cfg(cfg: Config):
             dataloader_cfg = Config(cfg_dict={**global_dataloader_cfg, **dataloader_cfg})
             cfg.data[f"{subset}_dataloader"] = dataloader_cfg
 
+    def configure_hooks(
+        self,
+        cfg,
+    ):
+        """Add or update hooks."""
+
+        if "custom_hooks" in self.override_configs:
+            override_custom_hooks = self.override_configs.pop("custom_hooks")
+            for override_custom_hook in override_custom_hooks:
+                update_or_add_custom_hook(cfg, ConfigDict(override_custom_hook))
+        if len(self.override_configs) > 0:
+            logger.info(f"before override configs merging = {cfg}")
+            cfg.merge_from_dict(self.override_configs)
+            logger.info(f"after override configs merging = {cfg}")
+
+        # add Cancel training hook
+        update_or_add_custom_hook(
+            cfg,
+            ConfigDict(type="CancelInterfaceHook", init_callback=self.on_hook_initialized),
+        )
+        if self.time_monitor is not None:
+            update_or_add_custom_hook(
+                cfg,
+                ConfigDict(
+                    type="OTXProgressHook",
+                    time_monitor=self.time_monitor,
+                    verbose=True,
+                    priority=71,
+                ),
+            )
+        cfg.log_config.hooks.append({"type": "OTXLoggerHook", "curves": self.learning_curves})
+        if hasattr(cfg, "algo_backend"):
```

Comment:
It's just a question. Why is it executed only if cfg has a `algo_backend`?

Reply:
It's just for safe. Updating caching modules require attributes from algo_backend in configuration.yaml file